solr/core/src/java/org/apache/solr/cloud/api/collections/DeleteCollectionCmd.java

```diff
@@ -116,6 +116,7 @@ public void call(ClusterState state, ZkNodeProps message, NamedList<Object> resu
       }
       ModifiableSolrParams params = new ModifiableSolrParams();
       params.set(CoreAdminParams.ACTION, CoreAdminParams.CoreAdminAction.UNLOAD.toString());
+      params.set(CoreAdminParams.DELETE_INDEX, message.getBool(CoreAdminParams.DELETE_INDEX, true));
```

Comment:
what is for?

Reply:
This fixes a bug that is latent in upstream, but that our code depends on. Everywhere else that cores are deleted, the index dir is explicitly called to be deleted. This often has no practical effect, because the index dir is usually a subdir of the data dir, so it gets deleted implicitly. But because the index dir is tracked independently, it's clearly bad hygiene to not do this. It could probably just be hardcoded to `true` (as the params set immediately below), but here for now I opted to allow explicit override to preserve the previous behavior exactly.

Reply:
Without this change, the in-memory tracking of the directory is not closed out, which is a bit sloppy, but doesn't present a practical problem in most cases (because the in-memory tracking is about to get GC'd anyway). But we need to clean up extra associated paths (the access dir) so we rely on this hook (which, again, should really be there anyway in principle).